src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -428,6 +467,100 @@ void AccessControlAttribute::OnEntryChanged(const SubjectDescriptor * subjectDes
     ChipLogError(DataManagement, "AccessControlCluster: event failed %" CHIP_ERROR_FORMAT, err.Format());
 }
 
+#if CHIP_CONFIG_USE_ACCESS_RESTRICTIONS
+CHIP_ERROR AccessControlAttribute::ReadCommissioningArl(AttributeValueEncoder & aEncoder)
+{
+    auto accessRestrictionProvider = GetAccessControl().GetAccessRestrictionProvider();
+    if (accessRestrictionProvider == nullptr)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+
+    return aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+        auto entries = accessRestrictionProvider->GetCommissioningEntries();
+
+        for (auto & entry : entries)
+        {
+            ArlStorage::EncodableEntry encodableEntry(entry);
+            ReturnErrorOnFailure(encoder.Encode(encodableEntry));
```

Comment:
This looks wrong.  `ArlStorage::EncodableEntry` encodes a fabric-scoped struct, but CommissioningARL consists of non-fabric-scoped structs.